### DISCUSSION TITLE
Home Assistant & MQTT - Show current data on screen?

### DISCUSSION THREAD

**Discussion by heijjan** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#448] [Ideas]
Everyhting works nicely now with Helsinki & MIAMI2021 controls. I have had a couple times a clogged filter, which causes E02 error on the display and prevents the pump to start. I can't see any information of this error in the MQTT section / Home Assistant sensors, could it be possible to include the data, what's displayed on the screen or include some signal for the errors so that I could make HA to report the issues when they occur?

**Comment by visualapproach** [comment]
For the moment you can track it by making sensor/sensors for char1-3.
Value_template: "{{ value_json.CH1 }}"
Value_template: "{{ value_json.CH2 }}"
Value_template: "{{ value_json.CH3 }}"
or adding them and check for "E02". Or just check CH1=='E'.

**Reply by heijjan** [reply]
Thanks, I added the following to the configuration.yaml:

`mqtt:
  sensor:
    name: "LayZSpa Display"
    state_topic: "layzspa/message"
    value_template: "{{ value_json.CH1 }}{{ value_json.CH2 }}{{ value_json.CH3 }}"
    availability_topic: "layzspa/Status"
    payload_available: "Alive"
    payload_not_available: "Dead"`

And now I get the value for the sensor "325055" so CH1 gives 32, CH2 gives 50 and CH3 gives 55. Should these values be decoded somehow? I'm quite sure that the display shows at the moment "27" as the temperature of the water.

**Reply by visualapproach** [reply]
Yes, it's ASCII values. I'm not a YAML ninja but maybe a search for "YAML int to char" will give you something. I don't have time right now to do it myself.

**Reply by heijjan** [reply]
Right, thanks. I guess in this case it is enough to do the conversion for the two latter channels so that it gives the ascii - 48, which equals to the number and then add a condition for the char 69, which equals to E for the channel 1. 

`mqtt:
  sensor:
    name: "LayZSpa Display"
    state_topic: "layzspa/message"
    value_template: >-
      {% if value_json.CH1 == 69 %}
      E{{ value_json.CH2 -48 }}{{ value_json.CH3 - 48 }}
      {% else %}
      {{ value_json.CH2 - 48 }}{{ value_json.CH3 - 48 }}
      {% endif %}
    availability_topic: "layzspa/Status"
    payload_available: "Alive"
    payload_not_available: "Dead"`

This should give the sensor the temperature value in normal mode and when in error, it should add letter E in front of the two digits. Now I can use that as a trigger for a notification automation.

**Reply by visualapproach** [reply]
Good job!

**Comment by dwoosnam** [comment]
I had this issue back last year @heijjan and solved it with [this mqtt template sensor](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/discussions/52#discussioncomment-2974599).

I really should look at updating my install to take advantage of the new features.

**Reply by heijjan** [reply]
Great, thank you! That is cleaner decoder than mine and it handles the characters correctly. The main thing is to be able to catch the E02 message for the clogged filter.
